packages/floating/src/hooks/useFloatingToolbar.ts

```diff
@@ -67,43 +72,62 @@ export const useFloatingToolbarState = ({
 export const useFloatingToolbar = ({
   editorId,
   floating,
-  focused,
   focusedEditorId,
   hideToolbar,
   ignoreReadOnly,
+  mouseupped,
   open,
   selectionExpanded,
   selectionText,
+  setMouseupped,
   setOpen,
   setWaitForCollapsedSelection,
   waitForCollapsedSelection,
 }: ReturnType<typeof useFloatingToolbarState>) => {
   // On refocus, the editor keeps the previous selection,
   // so we need to wait it's collapsed at the new position before displaying the floating toolbar.
   React.useEffect(() => {
-    if (!focused || ignoreReadOnly) {
+    if (!(editorId === focusedEditorId) || ignoreReadOnly) {
       setWaitForCollapsedSelection(true);
     }
     if (!selectionExpanded) {
       setWaitForCollapsedSelection(false);
     }
   }, [
-    focused,
+    editorId,
+    focusedEditorId,
     ignoreReadOnly,
     selectionExpanded,
     setWaitForCollapsedSelection,
   ]);
 
+  React.useEffect(() => {
```

Comment:
@felixfeng33 
This change prevents automatic testing tools from selecting text and displaying the floating toolbar.
We use playwright with text locators, for example:
`await this.page.getByText('Random text').selectText()`, these locators had the effect of making the floating toolbar appear but now this is no longer the case.

I tried to trigger the `mouseup` event manually from playwright but without success.

Why do we have to listen to the mouseup event in this hook?
Do you have a workaround for my use case ?

Reply:
listen to the mouseup event because we just need open the toolbar when the selection is end.

Reply:
sorry I'm not very familiar with Playwright.

Reply:
@felixfeng33 It looks like this change also breaks the toolbar for keyboard users. Can we change it to only suppress the toolbar if the pointer is currently down and the toolbar was hidden before pointerdown?

Reply:
Do you mean that the toolbar doesn't appear when you use shift to select? This is a known issue, and I am planning to fix it.

Reply:
That's right. It seems like making the dependency on specific input events optional (by waiting for pointerdown before suppressing the toolbar) would fix both the keyboard accessibility problem and @seb-mynotary's Playwright problem.